Updated time duration to 30 seconds

yoyounik opened the pull request:
![Screenshot 2024-10-22 150434](https://github.com/user-attachments/assets/59ece08e-e27e-408a-87de-d7c393f7f4cd)
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->

### What changes are being made and why?

<!-- Please include a brief summary of the changes included in this PR e.g. closes #1234. -->

---

### How the changes have been QAed?

<!-- Include example code that shows how this PR has been QAed. The code should present a complete yet easily reproducible flow.

```yaml
# Your example flow code here
```

Note that this is not a replacement for unit tests but rather a way to demonstrate how the changes work in a real-life scenario, as the end-user would experience them.

Remove this section if this change applies to all flows or to the documentation only. -->

---

### Setup Instructions

<!--If there are any setup requirements like API keys or trial accounts, kindly include brief bullet-points-description outlining the setup process below.

- [External System Documentation](URL)
- Steps to set up the necessary resources

If there are no setup requirements, you can remove this section.

Thank you for your contribution. ❤️  -->

yoyounik committed:
Updated time duration to 30 seconds

Skraye commented:
Hello @yoyounik, can you tell us more about your PR ?

yoyounik commented:
hello @Skraye 
Sure! In this PR, I’ve added a sleep task with a 30-second duration to the test workflow. The purpose of this change is to simulate a wait time between tasks in the workflow to ensure smooth task transitions.
Specifically:
 1. Updated the duration flow to include the sleep task of 30seconds.
 2.This change is useful for testing scenarios where a delay is required between task executions.
 
 please let me know if  you need more details or have any questions! Thanks for reviewing.

Skraye reviewed (changes requested):
As discuss here https://github.com/kestra-io/kestra/issues/4879 the issue has been misunderstood
Please make the change and restore the test as it was

yoyounik committed:
added Wait.java class, wait-task-flow.yaml

yoyounik committed:
Merge branch 'develop' into add-sleep-task

yoyounik commented:
hello @Skraye
I've made some changes to this PR:

Added a new Java class: io.kestra.plugin.core.debug
Updated the YAML file: core/src/test/resources/flows/valids/wait-task-flow.yaml
The new Java class and YAML file are necessary for implementing the wait task functionality.

Note: I've temporarily ignored writing test cases to expedite the PR review process. I'll add comprehensive test cases in a follow-up commit once the core functionality is approved.

Please review these changes and let me know if you have any questions or require further clarification.

Skraye commented on core/src/main/java/io/kestra/plugin/core/debug/Wait.java:
```diff
@@ -0,0 +1,55 @@
+        @Example(
+            code = """
+                id: wait
+                type: io.kestra.plugin.core.debug.Wait
```
Task type is not the good one, please check the issue for the good task name

Skraye commented on core/src/test/resources/flows/valids/sleep.yml:
```diff
@@ -4,4 +4,4 @@ namespace: io.kestra.tests
     type: io.kestra.core.tasks.test.Sleep
-    duration: 1000
\ No newline at end of file
+    duration: 3000 #30 seconds updated
```
Restore this, change isn't needed

Skraye commented on core/src/test/resources/flows/valids/wait-task-flow.yaml:
```diff
@@ -0,0 +1,17 @@
+    commands:
+      - echo "Wait task completed"
+
+triggers:
```
Probably copy past, triggers part is not required, remove it

yoyounik replied:
Hii @Skraye 
Thank you for your review and feedback on it. I understand the points you've raised and will make the necessary changes as soon as possible.

Skraye reviewed (commented):
Closed to perfection! Small changes and we are good, you can add a test :)

yoyounik committed:
updated the duration and added all other changes except test

yoyounik committed:
added the waitTaskTest method for testing

yoyounik commented:
hey @anna-geller @Skraye 
I've added a new test for the wait task in SequentialTest.java.
While running the tests, I encountered some Gradle build warnings related to deprecated features. Here's the relevant part of the output:
    Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0
I'm not familiar with the specific configuration of the Kestra build process. Could you please provide some guidance on how to address these warnings?

Additionally, any feedback on the SequentialTest.java test case would be greatly appreciated.

Skraye commented on core/src/main/java/io/kestra/plugin/core/debug/Wait.java:
```diff
@@ -0,0 +1,55 @@
+        @Example(
+            code = """
+                id: wait
+                type: wait.task
```
This is not a valid type, the example won't work

Skraye commented on core/src/main/java/io/kestra/plugin/core/debug/Wait.java:
```diff
@@ -0,0 +1,55 @@
+package io.kestra.plugin.core.debug;
```
The package is not the good one, here is the one asked in the original issue : 
`io.kestra.plugin.core.flow.Sleep`
with `io.kestra.plugin.core.flow` being the package and `Sleep` being the class name
and this will form the task type

Skraye replied:
@yoyounik Please read this, better take your times to make correct changes once than doing it 10 times

yoyounik replied:
@Skraye  I apologize for the repeated mistakes. I will take my time to ensure that the changes are correct before submitting them again.

yoyounik replied:
Hi @Skraye 
Thanks for the feedback. You're right, "Sleep" might be a more descriptive name for the class.

I can rename the Wait.java class to Sleep.java and update the code accordingly. This would keep the file in the io.kestra.plugin.core.flow package.

Does that sound like a good approach?

Skraye replied:
This is exactly what's need to be done :+1:

yoyounik replied:
okay then i will rename it and update you @Skraye

yoyounik replied:
Hey @Skraye 
changes i made:
Wait.java changed to Sleep.java,
wait-task-flow.yml changed to sleep-task-flow.yml and changes inside it to Sleep.
sleep.yml's  type has been changed.

going to update these changes now, please go through this once, and thanks for the patience shown :).
Please update me if any changes required.

Skraye commented on core/src/test/resources/flows/sleep-task.yml:
```diff
@@ -0,0 +1,7 @@
+id: sleep-task-flow
+namespace: io.kestra.tests
+
+tasks:
+  - id: sleep
+    type: io.kestra.core.tasks.scripts.Bash
+    script: sleep 30
```
Can you remove this ? Never use

yoyounik replied:
done @Skraye

Skraye replied:
Delete the file, not just remove the content please

yoyounik replied:
hello @Skraye 
Updated the changes according to your review, Thanks alot for the review , if you have time to review it once more , please check and tell if there is need of any more modifications.

yoyounik committed:
removed sleep-task.yml changes

yoyounik committed:
removed the type with id: wait
                      type: io.kestra.plugin.core.debug.Wait
                      duration: "PT5S"

yoyounik committed:
removed the type with id: wait
                      type: io.kestra.plugin.core.debug.Wait
                      duration: "PT5S"

        Also updated the wait.java in the flow package

Skraye commented on core/src/main/java/io/kestra/plugin/core/flow/Wait.java:
```diff
@@ -0,0 +1,55 @@
+        @Example(
+            code = """
+                id: wait
+                type: io.kestra.plugin.core.debug.Wait
```
Wrong type

Skraye commented on core/src/test/resources/flows/valids/wait-task-flow.yaml:
```diff
@@ -0,0 +1,14 @@
+
+tasks:
+  - id: wait
+    type: io.kestra.plugin.core.debug.Wait
```
Wrong type

Skraye commented on core/src/test/resources/flows/sleep-task.yml:
Delete it

yoyounik replied:
done @Skraye, made all the changes specified.

yoyounik committed:
removed the yaml file update the type

yoyounik committed:
Wait.java changed to Sleep.java,
wait-task-flow.yml changed to sleep-task-flow.yml and changes inside it to Sleep.
sleep.yml's type has been changed.

Skraye committed:
fix(*): polishing

yoyounik commented:
Thanks again @Skraye and everyone for the guiding out throughout the work process. @Skraye do you have any suggestions or guidance on how I can contribute more to the project? I'm excited to keep learning and contributing wherever I can.

Skraye commented:
We have a full list that are good for learning to work on the project : 
https://github.com/kestra-io/kestra/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+no%3Aassignee

Pick a new one in the list, take your times to understand the issue correctly, and feel free to open another PR!

yoyounik commented:
Thank you! @Skraye  I'll take a look at the list and make sure to fully understand the issues before opening another PR :)
